### PR TITLE
Show better message in case the pull request is closed or un-mergeable

### PR DESCRIPTION
**Pull request by nghialv**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: dockers/actions-plan-preview/main.go [hunk 1]
```diff
@@ -56,6 +56,32 @@ func main() {
...
+		)
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		log.Printf("Successfully commented plan-preview result on pull request\n%s\n", *comment.HTMLURL)
+	}
+
+	if event.PRClosed {
+		doComment(failureBadgeURL + "It was unable to run plan-preview for a closed pull request.")
```
Comment (khanhtc1202) [inline dockers/actions-plan-preview/main.go:76]:
```suggestion
		doComment(failureBadgeURL + "Unable to run plan-preview for a closed pull request.")
```
nits (same with the others), but it's up to you 👍

**Comment by khanhtc1202** [comment]
With small nits 🙆‍♀️ 
/lgtm

File: dockers/actions-plan-preview/main.go (hunk 1 above)
Comment (nghialv) [reply dockers/actions-plan-preview/main.go:76]: 👍

File: dockers/actions-plan-preview/main.go [hunk 2]
```diff
@@ -56,6 +56,32 @@ func main() {
...
+		log.Printf("Successfully commented plan-preview result on pull request\n%s\n", *comment.HTMLURL)
+	}
+
+	if event.PRClosed {
+		doComment(failureBadgeURL + "Unable to run plan-preview for a closed pull request.")
+		return
+	}
+
+	if !event.PRMergeable {
+		doComment(failureBadgeURL + "It was unable to run plan-preview for an un-mergeable pull request. Please resolve the conficts and try again.")
```
Comment (nghialv) [inline dockers/actions-plan-preview/main.go:81]:
```suggestion
		doComment(failureBadgeURL + "Unable to run plan-preview for an un-mergeable pull request. Please resolve the conficts and try again.")
```

File: dockers/actions-plan-preview/main.go [hunk 3]
```diff
@@ -71,20 +97,20 @@ func main() {
...
-	if err != nil {
-		log.Fatal(err)
+	// Maybe the PR is already closed so Piped could not clone the source code.
+	if result.HasError() {
+		pr, err := getPullRequest(ctx, ghClient, event.Owner, event.Repo, event.PRNumber)
+		if err != nil {
+			log.Fatal(err)
+		}
+		if !pr.GetClosedAt().IsZero() {
+			doComment(failureBadgeURL + "It was unable to run plan-preview for a closed pull request.")
```
Comment (khanhtc1202) [inline dockers/actions-plan-preview/main.go:107]:
```suggestion
			doComment(failureBadgeURL + "Unable to run plan-preview for a closed pull request.")
```
unify with the others 🙏

Comment (nghialv) [reply dockers/actions-plan-preview/main.go:107]: 👍

**Comment by khanhtc1202** [comment]
thx
/lgtm

**Comment by nakabonne** [comment]
yep, got more kind ;)
/approve

**Pull request merged**